pkg/textparse/openmetricsparse.go

```diff
@@ -170,6 +178,34 @@ func (p *OpenMetricsParser) Metric(l *labels.Labels) string {
 	return s
 }
 
+// Exemplar writes the exemplar of the current sample into the passed
+// exemplar. It returns the string from which the metric was parsed.
+func (p *OpenMetricsParser) Exemplar(e *exemplar.Exemplar) string {
+	// Allocate the full immutable string immediately, so we just
+	// have to create references on it below.
+	s := string(p.exemplar)
+
+	e.Value = p.exemplarVal
+	if p.hasExemplarTs {
+		e.Ts = &p.exemplarTs
```

Comment:
Won't this be lost when we parse the next one?

Reply:
I do not follow. Can we have multiple exemplars per metric? If not this should be ok I think as the same applies to all the information stored in the parser, no?

Reply:
You're taking a pointer which is static relative to the parser.

Reply:
Ok, yes. That's right. Its a problem above for `p.Ts` too. Fixed both.